android/src/main/java/com/instabug/reactlibrary/utils/RNTouchedViewExtractor.java

```diff
@@ -0,0 +1,171 @@
+                groupsCount++;
+            }
+        }
+        if (labelsCount > 1 || groupsCount > 0) return new MultiLabelsExtractionStrategy();
+        if (labelsCount == 1) return new SingleLabelExtractionStrategy();
+        return new NoLabelsExtractionStrategy();
+    }
+
+    interface ReactButtonExtractionStrategy {
+        @Nullable
+        TouchedView extract(ReactViewGroup reactButton, TouchedView touchedView);
+    }
+
+    class MultiLabelsExtractionStrategy implements ReactButtonExtractionStrategy {
+        private final String MULTI_LABEL_BUTTON_PRE_STRING = "a button that contains \"%s\"";
+
+        @Override
+        @Nullable
+        public TouchedView extract(ReactViewGroup reactButton, TouchedView touchedView) {
+
+            touchedView.setProminentLabel(
+                    InstabugCore.composeProminentLabelForViewGroup(reactButton, MULTI_LABEL_BUTTON_PRE_STRING)
+            );
+            return touchedView;
+        }
+    }
+
+    class PrivateViewLabelExtractionStrategy implements ReactButtonExtractionStrategy {
+
+        private final String PRIVATE_VIEW_LABEL_BUTTON_PRE_STRING = "a button";
+
+        @Override
+        public TouchedView extract(ReactViewGroup reactButton, TouchedView touchedView) {
+            touchedView.setProminentLabel(PRIVATE_VIEW_LABEL_BUTTON_PRE_STRING);
+            return touchedView;
+        }
+    }
+
+    class SingleLabelExtractionStrategy implements ReactButtonExtractionStrategy {
+
+        @Override
+        public TouchedView extract(ReactViewGroup reactButton, TouchedView touchedView) {
+            ReactTextView targetLabel = null;
+            for (int index = 0; index < reactButton.getChildCount(); index++) {
+                View currentView = reactButton.getChildAt(index);
+                if (!(currentView instanceof ReactTextView)) continue;
+                targetLabel = (ReactTextView) currentView;
+                break;
+            }
+            if (targetLabel == null) return touchedView;
+
+            String labelText = getLabelText(targetLabel);
+            touchedView.setProminentLabel(InstabugCore.composeProminentLabelFor(labelText, false));
+            return touchedView;
+        }
+
+        @Nullable
+        private String getLabelText(ReactTextView textView) {
+            String labelText = null;
+            if (!TextUtils.isEmpty(textView.getText())) {
+                labelText = textView.getText().toString();
+            } else if (!TextUtils.isEmpty(textView.getContentDescription())) {
+                labelText = textView.getContentDescription().toString();
+            }
+            return labelText;
+        }
+    }
+
+    class NoLabelsExtractionStrategy implements ReactButtonExtractionStrategy {
+
+        @Override
+        public TouchedView extract(ReactViewGroup reactButton, TouchedView touchedView) {
+            touchedView.setProminentLabel(
+                    InstabugCore.composeProminentLabelFor(null, false)
```

Comment:
What does this line serve?

Reply:
I'll check if we need to remove this line and if it's redundant I'll post updates right away 🙏 .

Reply:
I've removed it so it only returns the `touchedView` as is

```
class NoLabelsExtractionStrategy implements ReactButtonExtractionStrategy {
        @Override
        public TouchedView extract(ReactViewGroup reactButton, TouchedView touchedView) {
            return touchedView;
        }
    }
```